test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
So, in the case of MacOS (Unixy), it still attempts the mlock() even if it doesn't have that much physical memory; it churns through the physical memory locking it. This happened to Uri on the mailing list, and I was able to replicate it on my 8GB Mac. The Mac basically hung, swapping memory in and out as the unit test attempted to lock physical memory. Once it determined it could no longer allocate the necessary memory, it errored out. Linux (not sure of the kernel version), fails immediately on the plain mlock() because of the lack of physical memory. I don't know any of the BSD's behavior. And so, because of that, it was decided to remove the test. I kept a commented version of the test explaining what can happen; because it does show that the infinite loop fix works.